docs/source/torch_apis_torchscript.rst

```diff
@@ -0,0 +1,115 @@
+- ``torch.distributed.rpc.remote()``
+    - Executing a remote call on worker and getting a Remote Reference ``RRef`` as return value.
+    - More deatils about its usage and examples can be found in :meth:`~torch.distributed.rpc.rpc_async`.
+
+.. _torch_apis_in_torchscript_async:
+
+Asynchronous Execution
+^^^^^^^^^^^^^^^^^^^^^^
+
+TorchScript allows creating asynchronous computation tasks to make better use
+of computation resources. This is done via supporting a list of APIs that are
+only usable within TorchScript:
+
+- ``torch.jit.fork()``
+    - Creates an asynchronous task executing func and a reference to the value of the result of this execution. fork will return immediately.
+    - Synonymous to ``torch.jit._fork``, which is only kept for backward compatibility reasons.
+    - More deatils about its usage and examples can be found in :meth:`~torch.jit.fork`.
+- ``torch.jit.wait()``
+    - Forces completion of a ``torch.jit.Future[T]`` asynchronous task, returning the result of the task.
+    - Synonymous to ``torch.jit._wait``, which is only kept for backward compatibility reasons.
+    - More deatils about its usage and examples can be found in :meth:`~torch.jit.wait`.
+
+
+.. _torch_apis_in_torchscript_annotation:
+
+Type Annotations
+^^^^^^^^^^^^^^^^
+
+TorchScript is statically-typed, it provides and supports a set of utilities to help annotate variables and attributes.:
+
+- ``torch.jit.annotate()``
+    - Provides a type hint to TorchScript where Python 3 style type hints do not work well.
+    - One common example is to annotate type for expressions like ``[]``. ``[]`` is treated as ``List[torch.Tensor]`` by default, when a different type is needed, one can use following code to hint TorchScript: ``torch.jit.annotate(List[int], [])``.
+- ``torch.jit.Attribute``
+    - Constructor of a wrapper class that indicates a value should be considered to be an attribute of certain type of owning class.
+    - Most attribute types can be inferred, so ``torch.jit.Attribute`` is usually not necessary.
+- ``torch.jit.Final``
+    - Similar to Python's ``typing.Final``, it can be used to construct type annotation like ``torch.jit.Final[T]`` to indicate attributes that are immutable outside of initializer.
+    - TODO: Insert link to ``torch.jit.Final`` in Typing System Part II doc.
+
+.. _torch_apis_in_torchscript_meta_programming:
+
+Meta Programming
+^^^^^^^^^^^^^^^^
+
+TorchScript provides a set of utilities to facilitate meta programming.
+
+- ``torch.jit.is_scripting()``
+    - Returns a boolean value indicating whether current program is compiled by ``torch.jit.script`` or not.
+    - When used in an ``assert`` or ``if`` statement, the scope or branch where ``torch.jit.is_scripting()`` evaluates to ``False`` is not compiled.
+- ``torch.jit.ignore``
+    - This decorator indicates to the compiler that a function or method should be ignored and left as a Python function.
```

Comment:
What happens when an ignored function is invoked during the execution of TorchScript? For example, does it trigger a runtime error?